plugin/src/infobox.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  PostMessageProps,
+  PluginExtensionInstance,
+  SavePublicSetting,
+} from "@web/extensions/infobox/types";
+
+import html from "../dist/web/infobox/core/index.html?raw";
+
+const reearth = (globalThis as any).reearth;
+
+reearth.ui.show(html);
+
+let sidebarId: string;
+const getSidebarId = () => {
+  if (sidebarId) return;
+  sidebarId = reearth.plugins.instances.find(
+    (instance: PluginExtensionInstance) => instance.extensionId === "sidebar",
+  )?.id;
+};
+getSidebarId();
+
+reearth.on("message", ({ action, payload }: PostMessageProps) => {
+  if (action === "getInEditor") {
+    reearth.ui.postMessage({
+      action: "getInEditor",
+      payload: reearth.scene.inEditor,
+    });
+  } else if (action === "savePublicSetting") {
+    getSidebarId();
+    if (!sidebarId) return;
+    reearth.plugins.postMessage(sidebarId, {
+      action: "savePublicSetting",
+      payload,
+    } as SavePublicSetting);
+    console.log(payload);
```

Comment:
There is some code here and there that is for devs, like this console log. If it is on purpose to help you out once the new layer system is merged and to finish the logic connecting the infobox to the new layer system, it is fine. I just want to make sure you are leaving it on purpose and not forgetting

Reply:
This is the only console.log left in code. Actually this part is already touching the communication part so it may update in the future. I leave it here for confirm the save button works well.

Reply:
Removed.